### ISSUE TITLE
expect.hasAssertions() / expect.assertions() - Does not work when using expect() bound to current test

### ISSUE THREAD

**Issue by ryanone** [vitest-dev/eslint-plugin-vitest#567]
Problem: If you have the `prefer-expect-assertions` rule enabled, and also have test like the following...

```JavaScript
it('my test description, ({ expect }) => {
  const a = 1;
  const b = 2;

  expect(sum(a, b)).toBe(a + b);
})
```

ESLint fails to detect that `expect.assertions()` or `expect.hasAssertions()` is not present. I have an example repo here: https://github.com/ryanone/vitest-eslint-bug

But ESLint works properly when using an anonymous function instead of an arrow function...

```JavaScript
it('my test description, function ({ expect }) {
  // Now ESLint properly complains
  const a = 1;
  const b = 2;

  expect(sum(a, b)).toBe(a + b);
})
```

* Look at `src/math.test.ts`
* There are two test cases. The first one is the one with the `expect()` method bound to the current context, but does not have a call to `expect.assertions()` or `expect.hasAssertions()`. Yet, `npm run lint` passes. The latter test case uses the `expect()` method imported from vitest, and has a call to `expect.hasAssertions()`. If removed, `npm run lint` will fail.

**Comment by ryanone** [comment]
cc @austintheriot
